### PR TITLE
Fix tables that contain non-ASCII characters

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Slightly alarmed by this code, but seems to fix https://github.com/elifesciences/issues/issues/4878.

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
As far as I can tell, it's converting HTML like `<table><tr><td>Cell one</td></tr></table>` to `øtableø<tr><td>Cell one</td></tr>ø/tableø` to then base64-encode the contents so they _don't_ go through a Markdown processor.

**Comment by thewilkybarkid** [comment]
The UTF-8 modifier (`u`) also works, but would still fail on content containing `ø`.

File: src/modules/jcms_admin/src/HtmlMarkdownSerializer.php [hunk 1]
```diff
@@ -100,7 +100,7 @@ private function preserveOutput(string $html, array $context = []) : string {
       }
     }
     $preserve = implode('', $lines);
-    return preg_replace_callback('~' . $bc . '(code|table)[^' . $bc . ']*' . $bc . '([^' . $bc . ']*)' . $bc . '/\1' . $bc . '~s', function ($matches) use ($bc, $encode) {
```
Comment (giorgiosironi) [inline src/modules/jcms_admin/src/HtmlMarkdownSerializer.php]: can't understand why `$bc` was `ø`, but ok

Comment (nlisgo) [reply src/modules/jcms_admin/src/HtmlMarkdownSerializer.php]: It was just an arbitrary character.

**Pull request merged**